src/client/mixins/lead-aware.ts

```diff
@@ -3,8 +3,10 @@ export class LeadAwareMixin {
   client: Marketo;
   leadDescription: any;
 
-  public async createOrUpdateLead(lead: Record<string, any>) {
-    return this.client.lead.createOrUpdate([lead], { lookupField: 'email' });
+  public async createOrUpdateLead(lead: Record<string, any>, partitionId: number = 1) {
+    const partitions = await this.client.lead.partitions();
+    const partition = partitions.result.find(option => option.id === partitionId);
```

Comment:
what if the `partitionId` doesn't match any partition? I think we should respond with a `fail`

Reply:
like `if partition is provided, but doesn't exist, throw. If it's not provided, then create with 'Default'`